Path: news.gmane.org!not-for-mail
From: Anton Blanchard <anton@samba.org>
Newsgroups: gmane.linux.debian.devel.glibc
Subject: Bug#339415: libc6: Seen also on powerpc
Date: Fri, 16 Dec 2005 19:37:48 +1100
Lines: 41
Approved: news@gmane.org
Message-ID: <20051216084953.4D2A0688B3__3458.70919195495$1134724854$gmane$org@ozlabs.org>
Reply-To: Anton Blanchard <anton@samba.org>, 339415@bugs.debian.org
NNTP-Posting-Host: main.gmane.org
Mime-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit
X-Trace: sea.gmane.org 1134724854 28516 80.91.229.2 (16 Dec 2005 09:20:54 GMT)
X-Complaints-To: usenet@sea.gmane.org
NNTP-Posting-Date: Fri, 16 Dec 2005 09:20:54 +0000 (UTC)
Original-X-From: bounce-debian-glibc=debian-glibc=m.gmane.org@lists.debian.org Fri Dec 16 10:20:52 2005
Return-path: <bounce-debian-glibc=debian-glibc=m.gmane.org@lists.debian.org>
Original-Received: from murphy.debian.org ([146.82.138.6])
	by ciao.gmane.org with esmtp (Exim 4.43)
	id 1EnBjc-0001Yh-Lr
	for debian-glibc@m.gmane.org; Fri, 16 Dec 2005 10:18:40 +0100
Original-Received: from localhost (localhost [127.0.0.1])
	by murphy.debian.org (Postfix) with QMQP
	id A6BA12F0E3; Fri, 16 Dec 2005 03:18:22 -0600 (CST)
Old-Return-Path: <debbugs@bugs.debian.org>
X-Original-To: debian-glibc@lists.debian.org
Original-Received: from spohr.debian.org (spohr.debian.org [140.211.166.43])
	by murphy.debian.org (Postfix) with ESMTP id 5464E2F0E0;
	Fri, 16 Dec 2005 03:18:22 -0600 (CST)
Original-Received: from debbugs by spohr.debian.org with local (Exim 4.50)
	id 1EnBjK-0002DD-Gf; Fri, 16 Dec 2005 01:18:22 -0800
X-Loop: owner@bugs.debian.org
Resent-From: Anton Blanchard <anton@samba.org>
Resent-To: debian-bugs-dist@lists.debian.org
Resent-Cc: GNU Libc Maintainers <debian-glibc@lists.debian.org>
Resent-Date: Fri, 16 Dec 2005 09:18:21 UTC
Resent-Message-ID: <handler.339415.B339415.113472300511581@bugs.debian.org>
X-Debian-PR-Message: report 339415
X-Debian-PR-Package: glibc
X-Debian-PR-Keywords: 
Original-Received: via spool by 339415-submit@bugs.debian.org id=B339415.113472300511581
          (code B ref 339415); Fri, 16 Dec 2005 09:18:21 UTC
Original-Received: (at 339415) by bugs.debian.org; 16 Dec 2005 08:50:05 +0000
Original-Received: from ozlabs.org ([203.10.76.45])
	by spohr.debian.org with esmtp (Exim 4.50)
	id 1EnBHw-0002y7-S5
	for 339415@bugs.debian.org; Fri, 16 Dec 2005 00:50:05 -0800
Original-Received: by ozlabs.org (Postfix, from userid 1010)
	id 4D2A0688B3; Fri, 16 Dec 2005 19:49:53 +1100 (EST)
Original-To: Debian Bug Tracking System <339415@bugs.debian.org>
X-Mailer: reportbug 3.17
X-Spam-Checker-Version: SpamAssassin 2.60-bugs.debian.org_2005_01_02 
	(1.212-2003-09-23-exp) on spohr.debian.org
X-Spam-Level: 
X-Spam-Status: No, hits=-8.0 required=4.0 tests=BAYES_00,HAS_PACKAGE 
	autolearn=no version=2.60-bugs.debian.org_2005_01_02
Resent-Date: Fri, 16 Dec 2005 01:18:22 -0800
X-Rc-Virus: 2005-11-10_01
X-Rc-Spam: 2005-09-11_01
X-Mailing-List: <debian-glibc@lists.debian.org> archive/latest/18153
X-Loop: debian-glibc@lists.debian.org
List-Id: <debian-glibc.lists.debian.org>
List-Post: <mailto:debian-glibc@lists.debian.org>
List-Help: <mailto:debian-glibc-request@lists.debian.org?subject=help>
List-Subscribe: <mailto:debian-glibc-request@lists.debian.org?subject=subscribe>
List-Unsubscribe: <mailto:debian-glibc-request@lists.debian.org?subject=unsubscribe>
Precedence: list
Resent-Sender: debian-glibc-request@lists.debian.org
Xref: news.gmane.org gmane.linux.debian.devel.glibc:15838
Archived-At: <http://permalink.gmane.org/gmane.linux.debian.devel.glibc/15838>

Package: libc6
Version: 2.3.5-8
Followup-For: Bug #339415


The following patch, taken from mainline glibc cvs fixes the problem for
me.

--- sysdeps/wordsize-32/Makefile~	2002-09-19 01:47:56.000000000 -0500
+++ sysdeps/wordsize-32/Makefile	2005-11-16 13:19:03.000000000 -0600
@@ -2,5 +2,6 @@
 ifeq (yes,$(build-shared))
 sysdep_routines += divdi3
 shared-only-routines += divdi3
+CPPFLAGS-divdi3.c = -Din_divdi3_c
 endif
 endif
--- sysdeps/wordsize-32/symbol-hacks.h~	2004-03-08 14:59:34.000000000 -0600
+++ sysdeps/wordsize-32/symbol-hacks.h	2005-11-16 13:18:25.000000000 -0600
@@ -22,7 +22,7 @@
    therefore we get PLTs.  Unnecessarily so.  Changing gcc is a big
    task which might not be worth it so we play tricks with the
    assembler.  */
-#if !defined __ASSEMBLER__ && !defined NOT_IN_libc && defined SHARED
+#if !defined __ASSEMBLER__ && !defined in_divdi3_c && !defined NOT_IN_libc && defined SHARED
 asm ("__divdi3 = __divdi3_internal");
 asm ("__udivdi3 = __udivdi3_internal");
 asm ("__moddi3 = __moddi3_internal");

-- System Information:
Debian Release: testing/unstable
  APT prefers unstable
  APT policy: (500, 'unstable')
Architecture: powerpc (ppc)
Shell:  /bin/sh linked to /bin/bash
Kernel: Linux 2.6.15-rc2-g5d240918
Locale: LANG=en_AU, LC_CTYPE=en_AU (charmap=ISO-8859-1)

-- no debconf information


